docs/spec/v1beta1/kustomization.md

```diff
@@ -803,6 +803,20 @@ labeling or annotating them with:
 kustomize.toolkit.fluxcd.io/substitute: disabled
 ``` 
 
+Substitution needs to be defined opt-in. If not at least a single variable or resource to substitute from is defined, no var substitution is performed. So even expressions which should evaluate to a default value won't be evaluated. To enable var substitution, just set at least one variable which will enable var substitution for the Kustomization, if you don't have any other variables on the Kustomization. For example:
+
+```
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
+kind: Kustomization
+metadata:
+  name: apps
+spec:
+  ...
+  postBuild:
+    substitute:
+      enabled: "true"
```

Comment:
Given this can be _any_ value, I would use something more explicitly unique, to not accidentally set something we do not want to set and overwrite people their value.

Reply:
Something like that?

Reply:
Perfect!